### PR TITLE
Integrate GitHub Actions

### PR DESCRIPTION
**Pull request by SagiPolaczek**
Added GitHub Actions integration (like the one we have in Fuse).

### Remarks:
- I used the exact same configurations as we use in Fuse.
- Currently **ALL THE ERRORS ARE BEING IGNORED**. So in the near future we should do the following:
  1. apply `black` on the entire repo. It may cause to a lot of conflicts so we should consider the timing as well.
  2. "to flake" manually (delete missing imports and staffs like that). should be really easy & quick once we apply `black`.
  3. To solve all the typing issues that we have in the repo.

### CODE REVIEW

File: requirements/requirements.txt [hunk 1]
```diff
@@ -17,9 +17,9 @@ pg8000
 ipykernel  # dev
 jupyter  # dev 
 transformers
-black
-flake8
-mypy
+black==22.3.0  # dev
```
Comment (SagiPolaczek) [inline requirements/requirements.txt:20]: Modified version to match the GitHub Action's ones

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+[tool.black]
+line-length = 120
+
+exclude = """
+    .
```
Comment (SagiPolaczek) [inline pyproject.toml:5]: (ignore all) TEMP

File: .flake8 [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    # line break before binary operator
+    W503,
+    # invalid escape sequence '\s'
+    W605
+    # '<X>' may be undefined, or defined from star imports: <Y>
+    F405
+
+# Files to exclude:
+exclude =
+    fusedrug
```
Comment (mosheraboh) [inline .flake8:29]: Lets list the files explicitly so at least new files will raise errors

File: .mypy.ini [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+ignore_missing_imports = True
+show_column_numbers = True
+check_untyped_defs = True
+follow_imports = skip
+
+exclude = (?x)(
+    ^(
+      ##### Ignore all in the beginning
+      run_all_unit_tests.py
+      | fusedrug/
```
Comment (mosheraboh) [inline .mypy.ini:17]: same here

File: .travis.yml [hunk 5]
```diff
@@ -19,11 +19,5 @@ install:
 script:
```
Comment (mosheraboh) [inline .travis.yml:20]: lets remove the file

File: pyproject.toml (hunk 2 above)
Comment (mosheraboh) [reply pyproject.toml:5]: Lets black everything in this PR?

File: requirements/requirements.txt (hunk 1 above)
Comment (mosheraboh) [reply requirements/requirements.txt:20]: @alex-golts , we should move those to dev requirements (especially if we are forcing specific version)

**Review by mosheraboh — COMMENTED** [review]
Thanks Sagi!

File: requirements/requirements.txt (hunk 1 above)
Comment (SagiPolaczek) [reply requirements/requirements.txt:20]: Cool, I agree

File: pyproject.toml (hunk 2 above)
Comment (SagiPolaczek) [reply pyproject.toml:5]: Mmmmm,
I can do so but maybe we shall wait for other PRs to be closed? So we will avoid conflicts.
(I'm suggesting to black everything in a different consecutive PR 🙂 )

**Review by mosheraboh — APPROVED** [review]
Looks good

**Pull request merged**